cdap-app-fabric/src/test/java/io/cdap/cdap/internal/app/runtime/monitor/DirectRuntimeRequestValidatorTest.java

```diff
@@ -226,11 +231,15 @@ public void testFetcher() throws BadRequestException {
                                                                           accessEnforcer, authenticationContext);
 
     // The first call should be hitting the run record fetching to fetch the run record.
-    validator.validate(programRunId, new DefaultHttpRequest(HttpVersion.HTTP_1_1, HttpMethod.GET, "/"));
+    RunRecordStatus runRecordStatus = validator.checkProgramRunStatus(programRunId, new DefaultHttpRequest(
+      HttpVersion.HTTP_1_1, HttpMethod.GET, "/"));
+    Assert.assertEquals(programRunStatus, runRecordStatus.getProgramRunStatus());
 
     // The second call will hit the runtime store, so it shouldn't matter what the run record fetch returns
     runRecordFetcher.setRunRecord(null);
-    validator.validate(programRunId, new DefaultHttpRequest(HttpVersion.HTTP_1_1, HttpMethod.GET, "/"));
+    runRecordStatus = validator.checkProgramRunStatus(programRunId, new DefaultHttpRequest(HttpVersion.HTTP_1_1,
+                                                                                           HttpMethod.GET, "/"));
+    Assert.assertEquals(programRunStatus, runRecordStatus.getProgramRunStatus());
   }
 
```

Comment:
Add a new unit test for stopping with payload

Reply:
Will be part of https://cdap.atlassian.net/browse/CDAP-18743

Reply:
add a TODO in code with jira number